arkane/encorr/corrTest.py

```diff
@@ -0,0 +1,123 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE #
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER      #
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING     #
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER         #
+# DEALINGS IN THE SOFTWARE.                                                   #
+#                                                                             #
+###############################################################################
+
+"""
+This script contains unit tests for the :mod:`arkane.encorr.corr` module.
+"""
+
+import unittest
+
+import numpy as np
+
+from arkane.encorr.corr import (assign_frequency_scale_factor,
+                                get_atom_correction,
+                                get_bac,
+                                )
+from arkane.exceptions import AtomEnergyCorrectionError
+from arkane.modelchem import LevelOfTheory, CompositeLevelOfTheory
+
+
+class TestCorr(unittest.TestCase):
+    """
+    A class for testing the functions in corr.py.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        cls.freq_lot = LevelOfTheory(method='wb97X-D3', basis='def2-TZVP', software='Q-Chem')
+        cls.energy_lot = LevelOfTheory(method='CCSD(T)-F12', basis='cc-pVDZ-F12', software='MOLPRO')
+        cls.composite_lot = CompositeLevelOfTheory(freq=cls.freq_lot, energy=cls.energy_lot)
+        cls.lot_nonexisting = LevelOfTheory('notamethod')
+
+    def test_get_atom_correction(self):
+        """
+        Test that AECs can be assigned.
+        It's possible these values are refit in the future so a loose tolerance
+        is used to just test that the values can be queried. 
+        """
+        atoms = {'H': 1}
+        aec = get_atom_correction(level_of_theory=self.freq_lot, atoms=atoms)
+        self.assertAlmostEqual(aec, 1524327, places=None, delta=1000)
+
+        with self.assertRaises(AtomEnergyCorrectionError):
+            aec = get_atom_correction(level_of_theory=self.freq_lot, atoms={'X': 1})
+
+    def test_get_bac(self):
+        """
+        Test that the BACs can be assigned.
+        It's possible these values are refit in the future so a loose tolerance
+        is used to just test that the values can be queried.
+        """
+        bonds = {'H-H': 1}
+        # https://github.com/ReactionMechanismGenerator/RMG-database/blob/main/input/reference_sets/main/Dihydrogen.yml#L153
+        CCCBDB_coords = np.array([[0, 0, 0],
+                                  [0, 0, 0.7414],
+                                  ])
+        nums = (1, 1)
+
+        # test Petersson BACs
+        bac_type = 'p'
+        bac = get_bac(level_of_theory=self.freq_lot, bonds=bonds,
+                      coords=CCCBDB_coords, nums=nums,
+                      bac_type=bac_type,
+        )
+        self.assertAlmostEqual(bac, 700, places=None, delta=100)
```

Comment:
Can you also add a comment here about updating the value? 
`# The value can be updated by BAC(self.freq_lot, bac_type=bac_type).get_correction(bonds=bonds, coords=CCCBDB_coords, nums=nums).value_si`

Or we can also consider replacing the numeric value with the above function call, since we are basically testing if `get_bac` can reproduce the same results from the above function call. (and the above function call is tested in `bacTest.py`)